sources/rules.c

```diff
@@ -797,6 +869,26 @@ __attribute__((hot)) int od_rules_merge(od_rules_t *rules, od_rules_t *src,
 		}
 	}
 
+	/* sort rules according order, leaving obsolete at the end of the list */
+	od_list_t **sorted =
+		(od_list_t **)calloc(sizeof(od_list_t *), src_length);
```

Comment:
it does not really matter, but
1) I think it's not necessary to cast calloc result to (od_list_t **)
2) usually first param of calloc is number of elements and second is the size of element